common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1317,7 +1317,7 @@ def _xmodule(self):
                     error_msg=exc_info_to_str(sys.exc_info())
```

Comment:
This code (XModule construction via `construct_xblock_from_class()`) is called after `bind_for_student()` for XModules. However, for XBlocks, there doesn't seem to be a "catch error after bind_for_student() and return an ErrorBlock instead" code. 🤔

Reply:
Can you think of somewhere that would blow up in our face?

Reply:
Hmm, thinking about this some more:
1. Some XModules have/had `init()` methods which read student data. `ProblemBlock` used to be [one](https://github.com/edx/edx-platform/blob/8795bb8/common/lib/xmodule/xmodule/capa_module.py#L98-L105). Unless an `XBlock` overrides `bind_for_student()`, errors can only arise in a handler or a view call. But that results in the standard 500 for both XModules and XBlocks.
2. If I recall correctly, `xmodule_instance.save()` used to raise database integrity errors very frequently because of concurrent saves once upon a time and most of the exceptions here used to be for that. There is a `self.save()` in `bind_for_student()` but there is no error checking around it at least in the method.

So can't think of a case where the lack of it should cause any issues which haven't been encountered before. 🤔